addons/l10n_vn_edi/models/account_move.py

```diff
@@ -0,0 +1,698 @@
+                # Mark the invoice as not paid
+                endpoint = 'InvoiceAPI/InvoiceWS/cancelPaymentStatus'
+                params = {
+                    'supplierTaxCode': invoice.company_id.vat,
+                    'invoiceNo': invoice.l10n_vn_edi_invoice_number,
+                    'strIssueDate': invoice._l10n_vn_edi_format_date(invoice.l10n_vn_edi_issue_date),
+                }
+
+            if endpoint and params:
+                _request_response, error_message = invoice._l10n_vn_edi_send_request(
+                    method='POST',
+                    endpoint=endpoint,
+                    params=params,
+                )
+
+                if error_message:
+                    raise UserError(error_message)
+
+                # Revert back to the sent state as the status is up-to-date.
+                invoice.l10n_vn_edi_state = 'sent'
+
+    def _need_cancel_request(self):
+        # EXTEND 'account'
+        return super()._need_cancel_request() or self.l10n_vn_edi_invoice_number
+
+    def _post(self, soft=True):
+        # EXTEND 'account'
+        posted = super()._post(soft)
+
+        # Ensure to tag the move as 'Ready to send' upon posting if it makes sense.
+        posted.filtered(
+            lambda invoice:
+                invoice.company_id.account_fiscal_country_id.code == 'VN'
+                and invoice.is_sale_document()
+                and not invoice.l10n_vn_edi_invoice_number
+                and not invoice.l10n_vn_edi_state
+        ).l10n_vn_edi_state = 'ready_to_send'
+
+        return posted
+
+    # API
+
+    def _l10n_vn_edi_check_invoice_configuration(self):
+        """ Some checks that are used to avoid common errors before sending the invoice. """
+        self.ensure_one()
+        company = self.company_id
+        commercial_partner = self.commercial_partner_id
+        errors = []
+        if not company.l10n_vn_edi_username or not company.l10n_vn_edi_password:
+            errors.append(_('Sinvoice credentials are missing on company %s.', company.display_name))
+        if not company.vat:
+            errors.append(_('VAT number is missing on company %s.', company.display_name))
+        if company.phone and not company.phone.replace(' ', '').isdecimal():
+            errors.append(_('Phone number for company %s must only contain digits.', company.display_name))
+        if commercial_partner.phone and not commercial_partner.phone.replace(' ', '').isdecimal():
+            errors.append(_('Phone number for partner %s must only contain digits.', commercial_partner.display_name))
+        if not self.l10n_vn_edi_invoice_symbol:
+            errors.append(_('The invoice symbol must be provided.'))
+        if self.l10n_vn_edi_invoice_symbol and not self.l10n_vn_edi_invoice_symbol.invoice_template_id:
+            errors.append(_('The invoice symbol\'s template must be provided.'))
```

Comment:
Everyone asks different string format when they review :D
So I just try to stay consistent and use the same one everywhere. Here I can change since it makes sense